File: src/main/java/ma/abdellahelmoutaouakil/distributeurautomatique/dtos/TransactionDTO.java
package ma.abdellahelmoutaouakil.distributeurautomatique.dtos;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.List;

@Data
@AllArgsConstructor
public class TransactionDTO {
    private Long id;
    private float insertedAmount;
    private float changeGiven;
    private String status;
    private List<TransactionItemDTO> items;
}

File: frontend/src/app/features/vending/models/transaction.dto.ts
import { ProductDTO } from './product.dto';

export interface TransactionDTO {
  id: number;
  insertedAmount: number;
  changeGiven: number;
  items: TransactionItemDTO[];
}


export interface ChangeBreakdownDTO {
  [coinValue: string]: number;
}

export interface FinalizeTransactionResponseDTO {
  transaction: TransactionDTO;
  change: ChangeBreakdownDTO;
  status: string;
}

export interface TransactionItemDTO {
  id: number;
  productId: number;
  quantity: number;
  product: ProductDTO;
}
